examples/common.c

```diff
@@ -182,6 +184,33 @@ static int match_int_internal(
 	return 1;
 }
 
+int match_bool_arg(int *out, struct args_info *args, const char *opt)
+{
+	const char *found = args->argv[args->pos];
+	char *neg_opt = NULL;
+	int res = 0;
+
+	if (strcmp(found, opt) == 0) {
+		*out = 1;
+		return 1;
+	}
+
+	neg_opt = malloc(strlen(opt) + 3);
+	strcpy(neg_opt, "--no-");
+	strcat(neg_opt, opt + 2);
```

Comment:
This seems a bit heavy-handed ;)

```
if (!strncmp(found, "--no-", strlen("--no-")) &&
    !strcmp(found + strlen("--no-"), opt + 2)) {
        *out = 0;
        res = 1;
}
```

This is untested, but something like that would work.

Reply:
Nice, that's soo much better !